src/schedule.rs

```diff
@@ -226,9 +226,16 @@ impl Schedule {
 
                             for second in self.seconds.ordinals().range(second_range).cloned() {
                                 let timezone = after.timezone();
-                                let candidate = timezone
+
+                                let maybe_candidate = timezone
                                     .ymd(year as i32, month, day_of_month)
-                                    .and_hms(hour, minute, second);
+                                    .and_hms_opt(hour, minute, second);
+
+                                let candidate = match maybe_candidate {
+                                    Some(dt) => dt,
+                                    None => continue 'day_loop
```

Comment:
Skipping a full day seems like it might be too much. Consider this example expression:

```
*  1,59  *  *  *  *  *
^     ^  ^  ^  ^  ^  ^-- year
|     |  |  |  |  +----- day of week
|     |  |  |  +-------- month
|     |  |  +----------- day of month
|     |  +-------------- hour
|     +----------------- minute
+----------------------- second
```

If the `Schedule` iterator encounters a daylight savings time shift at minute `59`, we'd only want to skip ahead by two minutes to try minute `1` in the next hour.

I think the most conservative approach to this is to just skip to the next second in the loop. This would be wasteful when skipping a full hour (we'd have to evaluate up to 3600 datetimes), but there may be other invalid time spans (leap seconds?) that are shorter in duration. I don't know the details well enough to optimize for it.